metricbeat/module/kubernetes/util/metrics_storage.go

```diff
@@ -0,0 +1,200 @@
+package util
+
+import (
+	"strings"
+	"sync"
+)
+
+type Metrics struct {
```

Comment:
Structs would be better to be defined on top of file and have docstrings that explain what is their usage and how those are combined/related with other structs. Also check naming to be as self-explanatory as possible.

Reply:
I could use some help with the naming. I've run out of ideas. 
I've used:
- Metric for the enum for each metric
- Metrics for a list of metrics that belong to the same `source` (eg. a container or a node)
- MetricsRepo is where I store all the metrics from different sources (eg. container or node)

Reply:
I wonder if you can define the "repo" like this:

```
type MetricsRepo struct {
	sync.RWMutex
	metrics *MetricsStore
}

// MetricsStore are indexed per NodeName
type MetricsStore struct {
	Store map[key]nodeStore
}

type nodeStore struct {
   nodeMetrics NodeMetrics
   containersStore map[key]containerMetrics // key is Pod name, which is unique per namespace
   podStore          map[key]podMetrics.           // key is Container name in combination with Pod name and namespace
}

type NodeMetrics struct {
        metricX int
        metricY int
}

type containerMetrics struct {
        metricZ int
        metricW int
}

type podMetrics struct {
        metricP int
        metricT int
}
```

With this you have easy hierarchical access to the stores and you can easily delete metrics for Pods and Containers whenever a node is deleted, instantly.
Also this would give faster access and control over the Repo without the need to find the type of the Metrics and do the distinction etc. 
I used some names for structs on top of my mind but feel free to refactor this also let me know if this applies or if I lose everything :) .